scripts/experiment_2/nbconverted/4_similarity_analysis_hausdorff.py

```diff
@@ -0,0 +1,153 @@
+
+
+# In[7]:
+
+
+# Permute simulated data
+simulated_data_file = os.path.join(
+    base_dir,
+    "data",
+    "simulated",
+    analysis_name,
+    "simulated_data.txt.xz")
+
+# Read in data
+simulated_data = pd.read_table(
+    simulated_data_file,
+    header=0, 
+    index_col=0,
+    sep='\t')
+
+simulated_data.head(10)
+
+shuffled_simulated_arr = []
+num_samples = simulated_data.shape[0]
+
+for i in range(num_samples):
+    row = list(simulated_data.values[i])
+    shuffled_simulated_row = random.sample(row, len(row))
+    shuffled_simulated_arr.append(shuffled_simulated_row)
+
+shuffled_simulated_data = pd.DataFrame(shuffled_simulated_arr, 
+                                       index=simulated_data.index, 
+                                       columns=simulated_data.columns)
+shuffled_simulated_data.head()
+
+
+# In[11]:
+
+
+# Hausdorff distance using permuted dataset (Negative control)
+permuted_dist = max(directed_hausdorff(original_data_df, shuffled_simulated_data)[0],
+           directed_hausdorff(batch_data_df, shuffled_simulated_data)[0])
+
+threshold = pd.DataFrame(
+    pd.np.tile(
+        permuted_dist,
+        (len(num_batches), 1)),
+    index=num_batches,
+    columns=['distance'])
+
+permuted_dist
+
+
+# In[9]:
+
+
+# Plot
+dist_per_batch_effect = pd.DataFrame({'num_batch_effects':num_batches, 
+                                     'distance': output_list
+                                    })
+
+ggplot(dist_per_batch_effect, aes(x='num_batch_effects', y='distance'))     + geom_line()     + geom_line(aes(x=num_batches, y='distance'), threshold, linetype='dashed')     + xlab('Number of Batch Effects')     + ylab('Dissimilarity (Hausdorff distance)')     + ggtitle('Difference across increasing batch effects')
+
+
+# In[12]:
+
+
+# Print distance
+dist_per_batch_effect = pd.DataFrame(data={'distance': output_list},
+                                     index=num_batches)
+dist_per_batch_effect
```

Comment:
hmmm.... seems weird that all batch effects are similarly dissimilar to batch_1. if random addition of batch effects were effective, this should at least have some variance, no?

Reply:
This result was also surprising to me.  While this experiment started out looking into alternative similarity metrics, halfway through, this experiment started to focus more on the definition of batch effects and so I did not look more into the interpretation of these alternative methods.  But that is something I will consider in the future

Reply:
How are you looking into the definition of batch effects in this experiment? Can you give me an example of different batch effects considered?

Reply:
Some different definitions we've tried:
1.  Increasing the number of genes that change between batches from 25% to 75%
2.  Decreasing the constant that is used to scale the gene expression
3.  Change all genes instead of a subset
4.  Change the variance used for the gaussian distribution that is used to shift the gene expression data

Reply:
This needs to be explicitly written out with results from each in a summary somewhere.